feat(windows): add Windows platform support with WinAPI implementation

nglmercer opened the pull request:
- Implement Windows-specific window management using WinAPI
- Add platform-specific dependencies in Cargo.toml
- Create windows.rs with WinAPI window creation and rendering
- Update main.rs to handle Windows platform conditionally
- Add Windows-specific documentation and installation guides
- Implement message processing loop for Windows

nglmercer committed:
feat(windows): add Windows platform support with WinAPI implementation

- Implement Windows-specific window management using WinAPI
- Add platform-specific dependencies in Cargo.toml
- Create windows.rs with WinAPI window creation and rendering
- Update main.rs to handle Windows platform conditionally
- Add Windows-specific documentation and installation guides
- Implement message processing loop for Windows

Brayan-724 commented on src/main.rs:
```diff
@@ -9,43 +13,66 @@ extern crate gdkx11;
 async fn main() {
     // default configuration is to join chat as anonymous.
-    let config = ClientConfig::default();
+    let config: ClientConfig<StaticLoginCredentials> = ClientConfig::default();
```
This is already infered
```suggestion
```

Brayan-724 commented on docs/WINDOWS_SPECS.md:
```diff
@@ -0,0 +1,455 @@
+
+## 📦 Dependencias
+
+### Dependencias del Sistema
+- Windows 10/11 (recomendado)
+- Windows 8.1+ (mínimo)
+- Visual Studio Build Tools 2019+ o Visual Studio Community
+- Windows SDK 10.0+
```
Is this tested and verified?

Brayan-724 commented on docs/WINDOWS_SPECS.md:
```diff
@@ -0,0 +1,455 @@
+- Visual Studio Build Tools 2019+ o Visual Studio Community
+- Windows SDK 10.0+
+
+### Dependencias de Rust
```
This is not needed. It is better to comment `Cargo.toml` or use `[target.*.dependencies]`
```suggestion
```

Brayan-724 commented on docs/WINDOWS_SPECS.md:
The most of this content is not needed, it is just commented code

Brayan-724 commented on docs/LINUX_SPECS.md:
Same as `docs/WINDOWS_SPECS.md`

Brayan-724 commented on docs/INSTALLATION.md:
```diff
@@ -0,0 +1,513 @@
+## 📋 Requisitos Generales
+
+### Rust
+- **Versión mínima**: Rust 1.70+
```
Is it tested and verified?

Brayan-724 commented on docs/INSTALLATION.md:
````diff
@@ -0,0 +1,513 @@
+### Clonar el Repositorio
+```bash
+# Clonar desde Git
+git clone <repository-url>
````
`<repository-url>`? just put the url

Brayan-724 commented on docs/INSTALLATION.md:
```diff
@@ -0,0 +1,513 @@
+cd overlay-native
+
+# O descargar y extraer ZIP
+# wget <repository-zip-url>
```
same as [above](https://github.com/Brayan-724/overlay-native/pull/1/files#r2335254570)

Brayan-724 commented on docs/INSTALLATION.md:
````diff
@@ -0,0 +1,513 @@
+# Debe mostrar solo DLLs del sistema (kernel32.dll, user32.dll, etc.)
+```
+
+## 📋 Checklist de Instalación
+
+### Pre-instalación
+- [ ] Sistema operativo compatible
+- [ ] Rust 1.70+ instalado
+- [ ] Herramientas de compilación instaladas
+- [ ] Dependencias del sistema instaladas
+
+### Compilación
+- [ ] Repositorio clonado/descargado
+- [ ] `cargo build` ejecuta sin errores
+- [ ] `cargo test` pasa todos los tests
+- [ ] `cargo run` inicia la aplicación
+
+### Verificación
+- [ ] Aplicación se conecta a Twitch
+- [ ] Ventanas aparecen al recibir mensajes
+- [ ] Transparencia funciona correctamente
+- [ ] Texto se renderiza correctamente
+- [ ] Barra de progreso se actualiza
+- [ ] Ventanas se cierran automáticamente
+
+### Distribución (Opcional)
+- [ ] Compilación release exitosa
+- [ ] Binario funciona sin dependencias externas
+- [ ] Documentación incluida
+- [ ] Paquete creado correctamente
````
Remove all the checklist please

Brayan-724 commented on docs/ARCHITECTURE.md:
This file is completely wrong

Brayan-724 commented on src/main.rs:
```diff
@@ -62,31 +89,62 @@ async fn main() {
     windows_count %= total_windows;
 
-    client.join("apika_luca".to_owned()).unwrap();
+    client.join("mictia00".to_owned()).unwrap();
```
The tests/default-value are made with my channel.

```suggestion
    client.join("apika_luca".to_owned()).unwrap();
```

Brayan-724 commented on README.md:
````diff
@@ -0,0 +1,122 @@
+
+```bash
+# Clonar el repositorio
+git clone <repository-url>
````
Same as [above](https://github.com/Brayan-724/overlay-native/pull/1/files#r2335254570)

Brayan-724 commented on README.md:
```diff
@@ -0,0 +1,122 @@
+- ✅ Renderizado de ventanas en Linux
+- ✅ Barra de progreso funcional
+- ✅ Gestión de memoria y limpieza
+- 🔄 Carga de emotes (en progreso)
```
In progress just for windows, linux has the implementation

Brayan-724 commented on README.md:
```diff
@@ -0,0 +1,122 @@
+- ✅ Barra de progreso funcional
+- ✅ Gestión de memoria y limpieza
+- 🔄 Carga de emotes (en progreso)
+- ⏳ Configuración por archivo
+- ⏳ Interfaz gráfica de configuración
```
Not planned
```suggestion
```

Brayan-724 commented on README.md:
```diff
@@ -0,0 +1,122 @@
+
+## 📖 Documentación
+
+- [Especificaciones Linux](docs/LINUX_SPECS.md) - Detalles técnicos para la implementación GTK
+- [Especificaciones Windows](docs/WINDOWS_SPECS.md) - Detalles técnicos para la implementación WinAPI
+- [Arquitectura del Código](docs/ARCHITECTURE.md) - Estructura y diseño del proyecto
+- [Guía de Instalación](docs/INSTALLATION.md) - Instrucciones de compilación por plataforma
```
Check this after modify/remove these files

nglmercer committed:
docs: update documentation for clarity and conciseness

- Simplify README.md with more focused feature descriptions
- Condense platform-specific docs (LINUX_SPECS.md, WINDOWS_SPECS.md) to key technical details
- Streamline INSTALLATION.md with essential setup steps
- Refactor ARCHITECTURE.md to highlight current implementation
